terraform/environments/digital-prison-reporting/modules/domains/replay-pipeline/pipeline.tf

```diff
@@ -10,8 +10,31 @@ module "replay_pipeline" {
   definition = jsonencode(
     {
       "Comment" : "Replay Pipeline Step Function",
-      "StartAt" : "Stop DMS Replication Task",
+      "StartAt" : "Deactivate Archive Trigger",
       "States" : {
+        "Deactivate Archive Trigger" : {
+          "Type" : "Task",
+          "Resource" : "arn:aws:states:::glue:startJobRun.sync",
+          "Parameters" : {
+            "JobName" : var.glue_trigger_activation_job,
+            "Arguments" : {
+              "--dpr.glue.trigger.name" : var.archive_job_trigger_name,
+              "--dpr.glue.trigger.activate" : "false"
+            }
+          },
+          "Next" : "Stop Archive Job"
+        },
+        "Stop Archive Job" : {
```

Comment:
Any running instance of the archive job which might have started before the trigger deactivation is then stopped.